pyproject.toml

```diff
@@ -19,7 +19,7 @@ classifiers = [
     "Programming Language :: Python :: 3.12",
 ]
 dependencies = [
-    "asdf>=2.15.1,<4",
+    "asdf>=3.1.0,<4",
```

Comment:
Is there a strong reason for the `asdf` version requirement bump in this PR?

Reply:
This is required in order to allow using the `no:legacypath` option, because otherwise there are related failures inside asdf.  This was fixed in asdf by [this PR](https://github.com/asdf-format/asdf/pull/1759).